crates/biome_js_analyze/src/analyzers/nursery/no_done_callback.rs

```diff
@@ -0,0 +1,161 @@
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// beforeEach(done => {
+    ///     // ...
+    /// });
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// test('myFunction()', done => {
+    ///     // ...
+    /// });
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// beforeEach(async () => {
+    ///     // ...
+    /// });
+    /// ```
+    ///
+    /// ```js
+    /// test('myFunction()', () => {
+    ///     expect(myFunction()).toBeTruthy();
+    /// });
+    /// ```
+    ///
+    pub NoDoneCallback {
+        version: "next",
+        name: "noDoneCallback",
+        recommended: true,
+        source: RuleSource::EslintJest("no-done-callback"),
+        source_kind: RuleSourceKind::Inspired,
+    }
+}
+
+impl Rule for NoDoneCallback {
+    type Query = Ast<JsCallExpression>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        let callee = node.callee().ok()?;
+
+        let is_test_each = callee
+            .get_callee_member_name()
+            .map_or(false, |m| m.text_trimmed() == "each");
+
+        if is_test_each && !JsTemplateExpression::can_cast(callee.syntax().kind()) {
+            return None;
+        }
+
+        let arguments = &node.arguments().ok()?;
+        let callee_name = callee.get_callee_object_name()?;
+
+        let argument_index = match callee_name.text_trimmed() {
+            "after" | "afterAll" | "afterEach" | "before" | "beforeAll" | "beforeEach" => 0,
+            "it" | "test" => 1, // for test.each() and test() we want the second argument
+            _ => return None,
+        };
+        let argument = arguments
+            .get_arguments_by_index([argument_index])
+            .first()?
+            .clone();
+
+        let callback = argument?;
+        let callback = callback.as_any_js_expression()?;
+
+        match callback {
+            AnyJsExpression::JsArrowFunctionExpression(arrow_function) => {
+                let parameter = arrow_function.parameters().ok()?;
+                match parameter {
+                    AnyJsArrowFunctionParameters::AnyJsBinding(binding) => {
+                        let param = binding.as_js_identifier_binding()?;
+                        let text_range = param.name_token().ok()?;
+                        let text_range = text_range.text_trimmed_range();
+                        return Some(text_range);
+                    }
+                    AnyJsArrowFunctionParameters::JsParameters(js_parameters) => {
+                        if let Some(text_range) =
+                            analyze_js_parameters(&js_parameters, is_test_each)
+                        {
+                            return Some(text_range);
+                        }
```

Comment:
I noticed we can simply return the result.
```suggestion
                        return analyze_js_parameters(&js_parameters, is_test_each)
```